src/modules/rlm_kafka/rlm_kafka.c

```diff
@@ -820,15 +984,26 @@ static rlm_rcode_t CC_HINT(nonnull) kafka_common(void *instance, REQUEST *reques
 		return RLM_MODULE_FAIL;
 	}
 
-	if (radius_axlat(&key, request, section->key, NULL, NULL) < 0) {
-		RDEBUG3("Failed to expand key");
-		talloc_free(message);
-		return RLM_MODULE_FAIL;
+	if (section->key) {
+		if (radius_axlat(&key, request, section->key, NULL, NULL) < 0) {
+			RDEBUG3("Failed to expand key");
```

Comment:
Should this (and the one a few lines below) also be debug level 2? (and an error?)

Reply:
Agreed.